### ISSUE TITLE
Merge report

### ISSUE THREAD

**Issue by xNok** [toumorokoshi/deepmerge#5]
I think it will be nice to be able to be able to generate a report of which keys have been change as a result of the merge.

I don't known the best way to implement such a feature, so I open this ussie to talk about it.

There is three element to collect:
* path
* inital value
* new value

We don't want the implement something in each strategy so the solution sould be able to intercept the deepest change (in terms of path) and printing it.

**Comment by toumorokoshi** [comment]
Yeah, this may be difficult. An audit report would be cool: I would imagine you could feed in a data structure to store such merge information alongside the object you want to merge, that is responsible for collecting that data.

I think as long as performance isn't affected in the case you *don't* want a report, I'm happy to take a PR and merge it in.

**Comment by szepesisz** [comment]
If checking after the merge is sufficient, this could work:

https://python-json-patch.readthedocs.io/en/latest/tutorial.html#creating-a-patch

(With the limitation of being only able to process JSON-compatible types)

Edit:
After some searching, came across with this: https://github.com/seperman/deepdiff
Seems to be able to diff arbitrary objects
